StudentOrganisationForAerospaceResearch/OsirisPayload
# Integrated LPS22HH and MS5611 drivers into Barometer Task

The sensors module has been pulled up which allows us to integrate the 2 new barometer drivers. The barometers are initialized and setup with a resolution of 24 bits in the Barometer Task. Additionally, the barometer raw data is converted and used to report the current altitude, which is then broadcasted as a system message.

**Testing and Verification Instructions**
1. Build and flash the code onto the main board
2. Open a serial monitor and connect the PCB via a USB cable
3. Confirm that you are seeing the barometer readings being displayed in the serial terminal

**Integration Instructions**
If you want to obtain these barometer readings in your feature: 
1. Include the Barometer Task in your feature
2. Use the command router to subscribe to SYSTEM_BAROMETER_MESSAGES
3. Define a call back function that gets triggered when it receives the event SYSTEM_BAROMETER_MESSAGES

**Design Documentation**
Design diagrams for the class and message design can be found here:
https://www.freertos.org/Documentation/02-Kernel/02-Kernel-features/01-Tasks-and-co-routines/00-Tasks-and-co-routines
